core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1410,10 +1410,10 @@ class KafkaApis(val requestChannel: RequestChannel,
       // the group.instance.id field, so static members could accidentally become "dynamic", which leads to wrong states.
       sendResponseCallback(JoinGroupResult(
         List.empty,
-        JoinGroupResponse.UNKNOWN_MEMBER_ID,
+        JoinGroupRequest.UNKNOWN_MEMBER_ID,
```

Comment:
nit: why UNKNOWN_GENERATION_ID and UNKNOWN_PROTOCOL belongs to response while UNKNOWN_MEMBER_ID belongs to request? If there's no real good reason let's put them in a single class.

Reply:
I think it makes sense to move all the constant to request class.